src/utils.cpp

```diff
@@ -104,7 +104,9 @@ NumericVector gaussianized_data(NumericVector const& y,
   for (int i = 0; i < n; i++) {
     if (w(i) * exp(theta(i)) > 1e-3) {
       out(i) = y(i) * exp(-theta(i)) / w(i) - 1 + theta(i);
-    } else {  // deal with overflow using approximation
+    } else if (w(i) < 1e-10) { // deal with overflow using approximation
+      out(i) = y(i) - exp(theta(i)) / (w(i) + 1e-10) + theta(i);
```

Comment:
I actually suspect that 1e-10 may be too small. Perhaps, 1e-6?
```suggestion
    } else if (w(i) < 1e-6) { // deal with overflow using approximation
      out(i) = y(i) - exp(theta(i)) / 1e-6 + theta(i);
```